src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1129,6 +1146,20 @@ public void RegisterInputPorts()
         /// </summary>
         public void RegisterOutputPorts()
         {
+            // Old version of output ports registration.
+            // Used OutPortData.
+            if (OutPortData.Count > 0 && !isOutputPortsRegistered)
+            {
+                Warning(Properties.Resources.DeprecatedPortNamingStyleMessage, true);
+            }
+
+            // New version of output ports registration.
+            // Used port Attributes.
+            if (!isOutputPortsRegistered)
+            {
+                OutPortData.AddRange(GetPortsDataFromAttributes(PortType.Output));
```

Comment:
Can we use a local variable instead of mutating `OutPortData`? We should probably apply the `[Obsolete]` attribute to both the `InPortData` property and the `OutPortData` property.